### PR TITLE
Collapse seq

### PR DESCRIPTION
**Pull request by tissue3**
Implement sequence collapsing #74

### CODE REVIEW

**Comment by rachitnigam** [comment]
Hmm, `examples/ifseq.futil` still fails to compile. I would've expected it to compile with this.

File: calyx/src/passes/collapse_seq.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+///         (enable G H))
+/// ..)
+/// into 
+/// (seq (enable A B)
+///      (enable C D)
+///      (enable E F)
+///      (enable G H)
+///  ..)
+#[derive(Default)]
+pub struct CollapeSeq {}
```
Comment (rachitnigam) [inline calyx/src/passes/collapse_seq.rs]:
```suggestion
pub struct CollapseSeq {}
```

**Comment by tissue3** [comment]
> Hmm, `examples/ifseq.futil` still fails to compile. I would've expected it to compile with this.

Thanks for checking. I forgot to add main and mod

**Comment by rachitnigam** [comment]
That works! Merge it.

**Comment by rachitnigam** [comment]
Also, please run `rustfmt src/**/*.rs` to make sure all files are committed with the right formatting.

**Pull request merged**